File: AtCoder/ABS/python/ABC081B_Shift_only.py
N = int(input())
res = 1_000_000_001
A = [int(i) for i in input().split()]
for a in A:
    cnt = 0
    while a % 2 == 0:
        a >>= 1
        cnt += 1
    res = min(res, cnt)
print(res)

File: AtCoder/ABS/cpp/ABC081B_Shift_only.cpp
#include <algorithm>
#include <iostream>
#include <stdio.h>

#define MAX_A 1'000'000'001

using namespace std;

int main() {
  int N;
  cin >> N;
  int res = MAX_A;
  for (int i = 0; i < N; ++i) {
    int a;
    cin >> a;
    int cnt = 0;
    while (a % 2 == 0) {
      a >>= 1;
      ++cnt;
    }
    res = min(res, cnt);
  }
  printf("%d\n", res);
}